sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2902,6 +2907,10 @@ pi_result piKernelCreate(pi_program Program, const char *KernelName,
   } catch (...) {
     return PI_ERROR_UNKNOWN;
   }
+
+  // Update the refcount of the program to show its use by this kernel.
+  piProgramRetain(Program);
```

Comment:
Just a note: the program refcount will be 2 after piProgramCreate + piKernelCreate, while users are probably expecting 1. I believe OpenCL in this case keeps internal and external reference counting to not mix the two. I don't think we'd need to do this in this change-request though, if at all.

Reply:
Yes, this might be different than OpenCL.  I don't think we should add an internal ref_count to Program. If we thought that getting ref_count for program to be consistent with OpenCL, then I would simply recommend adding a field to Program, that would be exactly like a ref_count, but only for counting Kernels currently using this program.  That would have a kind of weird effect though.  It would mean that it would be possible to have a Program with a ref count of 0, but that hadn't actually been deleted yet because there were still kernels using it.

Reply:
The objects would get deleted when both user-visible refcount =0 AND internal refcounts all equal to 0 too. Anyways, let's just go with the single one for now.